test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js

```diff
@@ -634,8 +735,172 @@ describe( `[${ host }] Calypso Gutenberg Site Editor Tracking: (${ screenSize })
+						'Archive'
+					)
+				);
+				await editor.toggleNavigationSidebar();
+
+				const eventsStack = await getEventsStack( this.driver );
+				const editEvents = eventsStack.filter(
+					( [ eventName ] ) => eventName === 'wpcom_block_editor_nav_sidebar_item_add'
+				);
+				assert.strictEqual( editEvents.length, 1 );
+				const [ , editEventData ] = editEvents[ 0 ];
+				assert.strictEqual( editEventData.item_type, 'template' );
+				assert.strictEqual( editEventData.item_slug, 'archive' );
+			} );
+
+			it( 'should track "wpcom_block_editor_nav_sidebar_item_edit" when switching to a template part', async function () {
+				const editor = await SiteEditorComponent.Expect( this.driver );
+
+				await editor.toggleNavigationSidebar();
+				const backButtonToTemplatesLocator = driverHelper.createTextLocator(
+					By.css( '.components-navigation__back-button' ),
+					'Back'
+				);
+				await driverHelper.clickWhenClickable( this.driver, backButtonToTemplatesLocator );
+
+				await driverHelper.clickWhenClickable(
+					this.driver,
+					By.css(
+						'[role="region"][aria-label="Navigation Sidebar"] [role="menu"] [title="Template Parts"] button'
+					)
+				);
+				await driverHelper.clickWhenClickable(
+					this.driver,
+					By.css(
+						'[role="region"][aria-label="Navigation Sidebar"] [role="menu"] [title="Headers"] button'
+					)
+				);
+
+				const templateMenuItemLocator3 = driverHelper.createTextLocator(
+					By.css( '.edit-site-navigation-panel__template-item-title' ),
+					'header'
+				);
+				await driverHelper.clickWhenClickable( this.driver, templateMenuItemLocator3 );
+
+				const eventsStack = await getEventsStack( this.driver );
+				const editEvents = eventsStack.filter(
+					( [ eventName ] ) => eventName === 'wpcom_block_editor_nav_sidebar_item_edit'
+				);
+				assert.strictEqual( editEvents.length, 1 );
+				const [ , editEventData ] = editEvents[ 0 ];
+				assert.strictEqual( editEventData.item_type, 'template_part' );
+				assert.strictEqual( editEventData.item_id, 'pub/tt1-blocks//header' );
+			} );
+
+			it( 'make sure back to dashboard button exists', async function () {
+				const editor = await SiteEditorComponent.Expect( this.driver );
+
+				await editor.toggleNavigationSidebar();
+				const backButtonToTemplatesLocator = driverHelper.createTextLocator(
+					By.css( '.components-navigation__back-button' ),
+					'Template Parts'
+				);
+				await driverHelper.clickWhenClickable( this.driver, backButtonToTemplatesLocator );
+				const backButtonToTemplatesLocator2 = driverHelper.createTextLocator(
+					By.css( '.components-navigation__back-button' ),
+					'Back'
+				);
+				await driverHelper.clickWhenClickable( this.driver, backButtonToTemplatesLocator2 );
+
+				const isBackToDashboardLocated = await driverHelper.isElementLocated(
+					this.driver,
+					By.css( '.edit-site-navigation-panel__back-to-dashboard' )
+				);
+
+				assert( isBackToDashboardLocated );
```

Comment:
I think in this case it might be better to use `isElementEventuallyLocatedAndVisible`. It would provide the same result, but at the same time be less prone to flakiness as it would wait for the button in case it needs a sec to be mounted after the click:
```suggestion
				const isBackToDashboardLocated = await driverHelper.isElementEventuallyLocatedAndVisible(
					this.driver,
					By.css( '.edit-site-navigation-panel__back-to-dashboard' )
				);

				assert( isBackToDashboardLocated );
```